src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/AggregateCrossTargetProjectContextProvider.cs

```diff
@@ -241,6 +245,18 @@ private static string GetDisplayName(ConfiguredProject configuredProject, Projec
                 projectData.DisplayName;
         }
 
+        private Task ExecuteWithinLockAsync(Action action) => _gate.ExecuteWithinLockAsync(JoinableCollection, JoinableFactory, action);
+
+        private int _isInitialized;
+
+        private async Task EnsureInitialized()
```

Comment:
Would this be a good addition to `OnceInitializedOnceDisposedAsync`? Maybe push it down and make it `protected`, then remove the dupe from `UnconfiguredProjectContextProvider`

Also, should have `Async` suffix.